operator/webhook/introute/sync.py

```diff
@@ -111,15 +132,23 @@ def get_mounts(self) -> List[Mapping]:
                     "mountPath": cm_spec["mountPath"],
                 }
             )
-
         if self._tls_config:
-            volume_mounts.append(
-                {
-                    "name": self._tls_vol_name,
-                    "readOnly": True,
-                    "mountPath": TRUSTSTORE_PATH,
-                }
-            )
+            if self._tls_config['clientConfig']:
+                volume_mounts.append(
+                    {
+                        "name": self._tls_truststore_name,
+                        "readOnly": True,
+                        "mountPath": TRUSTSTORE_PATH,
+                    }
+                )
+            if self._tls_config['serverConfig']:
```

Comment:
Since `clientConfig` and `serverConfig` aren't marked as "required" in the CRD, accessing them directly with `[key]` could raise a `KeyError`. You might need to use `.get(key)` instead.